packages/ngtools/webpack/src/angular_compiler_plugin.ts

```diff
@@ -697,64 +716,66 @@ export class AngularCompilerPlugin {
-            this.options.additionalLazyModuleResources.includes(resource));
-
-        if (!result || !this.done || !isLazyModuleResource(result.resource)) {
-          return result;
-        }
-
-        return this.done.then(
-          () => {
-            // This folder does not exist, but we need to give webpack a resource.
-            // TODO: check if we can't just leave it as is (angularCoreModuleDir).
-            result.resource = path.join(this._basePath, '$$_lazy_route_resource');
-            // tslint:disable-next-line:no-any
-            result.dependencies.forEach((d: any) => d.critical = false);
-            // tslint:disable-next-line:no-any
-            result.resolveDependencies = (_fs: any, options: any, callback: Callback) => {
-              const dependencies = Object.keys(this._lazyRoutes)
-                .map((key) => {
-                  const modulePath = this._lazyRoutes[key];
-                  if (modulePath !== null) {
-                    const name = key.split('#')[0];
-
-                    return new this._contextElementDependencyConstructor(modulePath, name);
-                  } else {
-                    return null;
-                  }
-                })
-                .filter(x => !!x);
-
-              if (this._options.nameLazyFiles) {
-                options.chunkName = '[request]';
-              }
-
-              callback(null, dependencies);
-            };
-
+    if (this._discoverLazyRoutes) {
+      // Add lazy modules to the context module for @angular/core
+      compiler.hooks.contextModuleFactory.tap('angular-compiler', cmf => {
+        const angularCorePackagePath = require.resolve('@angular/core/package.json');
+
+        // APFv6 does not have single FESM anymore. Instead of verifying if we're pointing to
+        // FESMs, we resolve the `@angular/core` path and verify that the path for the
+        // module starts with it.
+        // This may be slower but it will be compatible with both APF5, 6 and potential future
+        // versions (until the dynamic import appears outside of core I suppose).
+        // We resolve symbolic links in order to get the real path that would be used in webpack.
+        const angularCoreResourceRoot = fs.realpathSync(path.dirname(angularCorePackagePath));
+
+        cmf.hooks.afterResolve.tapPromise('angular-compiler', async result => {
+          // Alter only existing request from Angular or the additional lazy module resources.
+          const isLazyModuleResource = (resource: string) =>
+            resource.startsWith(angularCoreResourceRoot) ||
+            ( this.options.additionalLazyModuleResources &&
+              this.options.additionalLazyModuleResources.includes(resource));
+
+          if (!result || !this.done || !isLazyModuleResource(result.resource)) {
             return result;
-          },
-          () => undefined,
-        );
+          }
+
+          return this.done.then(
```

Comment:
Nit: we can use await here to reduce a bit the indentations and nesting.

Reply:
Refactored to use await.